dmutils/timing.py

```diff
@@ -0,0 +1,82 @@
+from contextlib import contextmanager
+import logging
+import sys
+from time import perf_counter, process_time
+
+from flask import request
+from flask.ctx import has_request_context
+
+
+#
+# logged_duration's defaults are exposed here so that a caller is able to defer to the default or simply use a wrapper
+# for the default when customizing the call.
+#
+
+
+def default_message(log_context):
+    return "Block {} in {{duration_real}}s of real-time".format(
+        "executed" if sys.exc_info()[0] is None else "raised {}".format(sys.exc_info()[0].__name__)
+    )
+
+
+def default_condition(log_context):
+    return has_request_context() and getattr(request, "sampling_decision", False)
```

Comment:
`sampling_decision` feels like a weird phrase to me. Does it have an origin I'm not aware of or is it something you've come up with yourself?

Reply:
FWIW I've made a couple of suggestions on that PR.